### ISSUE TITLE
Extra Softmax layer in deep model

### ISSUE THREAD

**Issue by amir-haji** [sharif-ml-lab/IMDb-IR-System#13]
When the nn.CrossEntropyLoss is used as the criterion, regarding [pytorch documentation](https://pytorch.org/docs/stable/generated/torch.nn.CrossEntropyLoss.html), the loss fuction applies the softmax on the input. Therefore, there is no need to add an extra softmax layer at the end of the model.

https://github.com/sharif-ml-lab/IMDb-IR-System/blob/ab4e92b6957fe747331744d815946b4eb6b57d90/Logic/core/classification/deep.py#L28C1-L42C10?plain=1

**Comment by amirhossein-razlighi** [comment]
Hi. Thanks for your comment! You are absolutely right. Please make a corresponding PR to solve this issue.

**Comment by amir-haji** [comment]
Thanks. I'll create my PR soon.
